lib/rmt/cli/mirror.rb

```diff
@@ -70,34 +68,51 @@ def product(*targets)
         products = Product.get_by_target!(target)
         raise RMT::CLI::Error.new(_('Product for target %{target} not found') % { target: target }) if products.empty?
         products.each do |product|
-          product_repos = product.repositories.where(mirroring_enabled: true)
+          product_repos = product.repositories.only_mirroring_enabled
           raise RMT::CLI::Error.new(_('Product %{target} has no repositories enabled') % { target: target }) if product_repos.empty?
           repos += product_repos.to_a
         end
       rescue ActiveRecord::RecordNotFound
         raise RMT::CLI::Error.new(_('Product with ID %{target} not found') % { target: target })
       end
 
-      repos.each do |repo|
-        mirror_repo!(mirror, repo)
-      rescue RMT::Mirror::Exception => e
-        logger.warn e.to_s
-      end
+      _, repo_errors = mirror_repos!(mirror, repos)
+      handle_errors!(repo_errors)
     end
   end
 
   protected
 
-  def mirror_repo!(mirror, repo)
-    mirror.mirror(
-      repository_url: repo.external_url,
-      local_path: Repository.make_local_path(repo.external_url),
-      auth_token: repo.auth_token,
-      repo_name: repo.name
-    )
+  def mirror_repos!(mirror, repos)
```

Comment:
Do we still need to call this `mirror_repos!`? Maybe, no with it not raising errors anymore, we could call it `mirror_repos`.

Reply:
Previously, it was `mirror_repo!` and it had no error raising statements, my guess is it had the exclamation because it would modify the repository objects with `repo.refresh_timestamp!`.

I can change it.

Reply:
Wait, you may be right with the refresh_timestamp! You can ignore my comment ^_^